tests/components/group/test_binary_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""The tests for the Group Binary Sensor platform."""
+from os import path
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.group import DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, STATE_ON, STATE_UNAVAILABLE
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+
+async def test_default_state(hass):
+    """Test binary_sensor group default state."""
+    hass.states.async_set("binary_sensor.kitchen", "on")
+    hass.states.async_set("binary_sensor.bedroom", "on")
+    await async_setup_component(
+        hass,
+        BINARY_SENSOR_DOMAIN,
+        {
+            BINARY_SENSOR_DOMAIN: {
+                "platform": DOMAIN,
+                "entities": ["binary_sensor.kitchen", "binary_sensor.bedroom"],
+                "name": "Bedroom Group",
+                "unique_id": "unique_identifier",
+                "device_class": "presence",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.bedroom_group")
+    assert state is not None
+    assert state.state == STATE_ON
+    assert state.attributes.get(ATTR_ENTITY_ID) == [
+        "binary_sensor.kitchen",
+        "binary_sensor.bedroom",
+    ]
+
+    entity_registry = er.async_get(hass)
+    entry = entity_registry.async_get("binary_sensor.bedroom_group")
+    assert entry
+    assert entry.unique_id == "unique_identifier"
+
+
+async def test_state_reporting(hass):
```

Comment:
this tests the behavior with `all = True`, we should add one which tests the behavior with `all = False` too. The same improvement should be done for the other groups in a follow-up.

Reply:
Agreed. The odd thing about using the binary sensor base is the state is forced to be either true/false, and the unavailable state is removed, so you can't actually return unavailable if the member entities are unavailable. Anyway, happy to add the addition test case or cases if this PR has any hope of being merged.

Reply:
Availability is still supported, but it's done by implementing the availability property attribute, not by writing to the state directly.

Either:
```
@property
    def available(self) -> bool:
        """Return True if entity is available."""
        return ...
```

Or:
```py
self._attr_available = ...
```